git_machete/cli.py

```diff
@@ -75,274 +77,423 @@ def usage(command: str = None) -> None:
+            subparser_name = full_parser_name_splited[1]
+            usage(subparser_name)
+        else:
+            usage()
+        parser.exit()
+
+
+def create_cli_parser() -> argparse.ArgumentParser:
+    cli_parser = argparse.ArgumentParser(
+        prog='git-machete', argument_default=argparse.SUPPRESS, add_help=False)
+    cli_parser.add_argument('--debug', action='store_true')
+    cli_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+    cli_parser.add_argument(
+        '--version', action='version', version=f'%(prog)s version {__version__}')
+    cli_parser.add_argument('-v', '--verbose', action='store_true')
+
+    subparsers = cli_parser.add_subparsers(dest='command')
+
+    add_parser = subparsers.add_parser('add', argument_default=argparse.SUPPRESS, add_help=False)
+    add_parser.add_argument('branch', nargs='?')
+    add_parser.add_argument('-o', '--onto')
+    add_parser.add_argument('-R', '--as-root', action='store_true')
+    add_parser.add_argument('-y', '--yes', action='store_true')
+    add_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+
+    advance_parser = subparsers.add_parser('advance', add_help=False)
+    advance_parser.add_argument('-y', '--yes', action='store_true', default=argparse.SUPPRESS)
+    advance_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+
+    anno_parser = subparsers.add_parser('anno', add_help=False)
+    # possible values of 'annotation_text' include: [], [''], ['some_val'], ['text_1', 'text_2']
+    anno_parser.add_argument('annotation_text', nargs='*')
+    anno_parser.add_argument('-b', '--branch', default=argparse.SUPPRESS)
+    anno_parser.add_argument(
+        '-H', '--sync-github-prs', action='store_true', default=argparse.SUPPRESS)
+    anno_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+
+    delete_unmanaged_parser = subparsers.add_parser('delete-unmanaged', add_help=False)
+    delete_unmanaged_parser.add_argument(
+        '-y', '--yes', action='store_true', default=argparse.SUPPRESS)
+    delete_unmanaged_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+
+    diff_full_parser = subparsers.add_parser('diff', aliases=['d'], add_help=False)
+    diff_full_parser.add_argument('branch', nargs='?')
+    diff_full_parser.add_argument(
+        '-s', '--stat', action='store_true', default=argparse.SUPPRESS)
+    diff_full_parser.add_argument('-h', '--help', action=MacheteHelpAction)
+
+    discover_parser = subparsers.add_parser(
+        'discover', argument_default=argparse.SUPPRESS, add_help=False)
+    discover_parser.add_argument('-C', '--check-out-since')
+    discover_parser.add_argument('-l', '--list-commits', action='store_true')
+    discover_parser.add_argument('-r', '--roots')
+    discover_parser.add_argument('-y', '--yes', action='store_true')
+    discover_parser.add_argument('-h', '--help', action=MacheteHelpAction)
```

Comment:
Nit: is it possible to add `--help` argument to all subparses in one go/via some loop at the very end?

Reply:
I've done it via adding new common args parser and inheriting args from it. It should work now.